db.js

```diff
@@ -0,0 +1,118 @@
+require("dotenv").config()
+const { Client } = require("pg");
+
+const client = new Client({
+  user: process.env.DB_USER,
+  host: process.env.DB_HOST,
+  database: process.env.DB_NAME,
+  password: process.env.DB_PASSWORD,
+  port: Number(process.env.DB_USER),
+});
+
+client
+  .connect()
+  .then(() => console.log("Connected to database"))
+  .catch((err) => console.error("Error connecting to database: ", err.stack));
+
+async function closeConnection() {
+  client
+    .end()
+    .then(() => console.log("Connection ended"))
+    .catch((err) => console.error("Error closing connection: ", err.stack));
+}
+
+async function fetchTodos() {
+  try {
+    res = await client.query("SELECT * FROM todo");
+    if (res.rowCount === 0) {
+      console.log("No tasks found");
+      return;
+    } else {
+      res.rows.forEach((row) =>
+        console.log(`${row.id}. ${row.task} [${row.done ? "x" : ""}]`)
+      );
+    }
+  } catch (error) {
+    console.error("Error fetching todos: ", error.stack);
+  } finally{
+    closeConnection();
+  }
+}
+
+async function addTodo(values) {
+  try {
+    res = await client.query("INSERT INTO todo(task) VALUES($1) RETURNING *", [
+      values,
+    ]);
+    console.log(`Added new task: "${res.rows[0].task}"`);
+  } catch (error) {
+    console.error(error);
+  } finally{
+    closeConnection();
+  }
+}
+
+async function markTaskAsDone(id) {
+  try {
+    res = await client.query(
+      "UPDATE todo SET done = true WHERE id = $1 RETURNING *",
+      [id]
+    );
+    await client.query("UPDATE todo SET updatedAt = CURRENT_TIMESTAMP WHERE id = $1 RETURNING *", [id])
+    console.log(`Marked "${res.rows[0].task}" as done`);
```

Comment:
Can't all this be one query or a transaction?

Reply:
Yeah i think that would be better. Thanks